base_domain_inverse_function/inverse_expression.py

```diff
@@ -0,0 +1,90 @@
+# Copyright 2022 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+from odoo.osv.expression import (
+    AND,
+    AND_OPERATOR,
+    DOMAIN_OPERATORS,
+    NOT_OPERATOR,
+    OR,
+    OR_OPERATOR,
+)
+
+
+def inverse_combine(domain, operator):
```

Comment:
I find it very difficult to understand.
Domains use infix notation `a + b -> + b a`.
I remember in compilation courses at school, we created a lexer for that notation, and I would have gone this way.
The grammar here is quite easy: `|, &, ! and tuples`, I think it would be easier to review and to maintain.
However, it might take some time to implement.
Ping me if you have questions.

Reply:
Feel free to propose another implementation if you can do better :stuck_out_tongue_closed_eyes: